tick/stateful/functions.go

```diff
@@ -1354,6 +1355,41 @@ func (year) Signature() map[Domain]ast.ValueType {
 	return timeFuncSignature
 }
 
+var nowFuncSignature = map[Domain]ast.ValueType{}
+
+// Initialize Now function signature
+func init() {
+	d := Domain{}
+	nowFuncSignature[d] = ast.TTime
+}
+
+var timeNowFunc func() time.Time = time.Now
+
+func SetTimeNowFunc(newTimeNowFunc func() time.Time) {
```

Comment:
This function is only for testing right? In general, I'm not a huge fan of having package level functions like this. I think we should just get rid of this and use `time.Now` explicitly in `now.Call`.

Reply:
Yes, it's only for testing. I'm not fan of this way of doing things either, but without it, the tests will be reliant on a parameter (time) that's not deterministic. But I understand your point of view and I'll rework my PR.